src/v/hashing/utils.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "utils/functional.h"
```

Comment:
how do we avoid making v::utils a dependency of v::hashing? even though this is header-only, it's going to break when we flip the switch later where the dependencies have to be explicitly recorded.

public headers should go into hashing/include/hashing

Reply:
yup will just copy that , removing the include